secant/Features/Welcome/WelcomeView.swift

```diff
@@ -0,0 +1,113 @@
+    var body: some View {
+        ZStack {
+            GeometryReader { proxy in
+                let outterPadding = proxy.size.height * 0.015
+                let firstPadding = proxy.size.height * 0.075 + outterPadding
+                let secondRingPadding = firstPadding * 1.5
+                let outerShadowDrop = proxy.size.height * 0.14
+                let outerShadowOffset = proxy.size.height * 0.055
+
+                Circle()
+                    .fill(
+                        LinearGradient(
+                            colors: [
+                                Asset.Colors.ZcashBadge.outerRingGradientStart.color,
+                                Asset.Colors.ZcashBadge.outerRingGradientEnd.color
+                            ],
+                            startPoint: UnitPoint(x: 0.5, y: 0),
+                            endPoint: UnitPoint(x: 0.5, y: 1)
+                        )
+                    )
+                    .if(colorScheme == .light) { view in
+                        view.shadow(
+                            color: Asset.Colors.ZcashBadge.shadowColor.color,
+                            radius: outerShadowDrop,
+                            x: outerShadowOffset,
+                            y: outerShadowOffset
+                        )
+                    }
+
+                Circle()
+                    .foregroundColor(Asset.Colors.ZcashBadge.thickRing.color)
+                    .padding(outterPadding)
+
+                Circle()
+                    .foregroundColor(Asset.Colors.ZcashBadge.thinRing.color)
+                    .padding(firstPadding)
+                Circle()
+                    .foregroundColor(Asset.Colors.ZcashBadge.innerCircle.color)
+                    .padding(secondRingPadding)
+
+                ZcashSymbol()
+                    .fill(Asset.Colors.ZcashBadge.zcashLogoFill.color)
+                    .padding(firstPadding + secondRingPadding)
+            }
+        }
+    }
+}
+
+struct WelcomeView_Previews: PreviewProvider {
+    static let squarePreviewSize: CGFloat = 360
+    static var previews: some View {
+        // These lines are commented out because Xcode goes centennial and says "preview took more than 5 seconds"
+
+        //        ZcashBadge()
+        //            .applyScreenBackground()
+        //            .previewLayout(.fixed(width: squarePreviewSize, height: squarePreviewSize))
+        //            .preferredColorScheme(.dark)
+        //        ZStack {
+        //            ZcashBadge()
+        //        }
+        //        .padding()
+        //        .applyScreenBackground()
+        //        .previewLayout(.fixed(width: squarePreviewSize, height: squarePreviewSize))
+        //        .preferredColorScheme(.light)
+
+        Group {
+            WelcomeView()
+                .preferredColorScheme(.dark)
+            WelcomeView()
+                .previewDevice("iPhone SE (2nd generation)")
+        }
+    }
+}
```

Comment:
I was able to get these previews to work by commenting out each one, and letting the previews generate one at a time, and eventually having all of them display together just fine.

Not sure if it's related to formatting and spacing but I made the following formatting changes so that you can read the code easier with the previews open in Xcode.

```suggestion
    static var previews: some View {
        ZcashBadge()
            .applyScreenBackground()
            .previewLayout(
                .fixed(
                    width: squarePreviewSize,
                    height: squarePreviewSize
                )
            )
            .preferredColorScheme(.dark)

        ZStack {
            ZcashBadge()
        }
        .padding()
        .applyScreenBackground()
        .previewLayout(
            .fixed(
                width: squarePreviewSize,
                height: squarePreviewSize
            )
        )
        .preferredColorScheme(.light)

        Group {
            WelcomeView()
                .preferredColorScheme(.dark)

            WelcomeView()
                .previewDevice("iPhone SE (2nd generation)")
        }
    }
```

Reply:
Whoa! this appears to work.

Reply:
I wouldn't be surprised if it was a formatting issue.  It's possible that it thought you were trying to nest the zstack or something.